### ISSUE TITLE
support libcamera

### ISSUE THREAD

**Issue by pdp7** [starfive-tech/beagle_kernel_5.10#21]
It is ideal if the JH7100 video and camera input (CSI + ISP) are able to be supported by libcamera
https://libcamera.org/
@kbingham can assist with but coordination with StarFive engineers is needed in order to do do

**Comment by tekkamanninja** [comment]
we will try to build this rpm package to work with the latest JH7100 video and camera driver(V4L2).

**Comment by kbingham** [comment]
Is support for this platform still ongoing?

**Comment by MichaelZhuxx** [comment]
> Is support for this platform still ongoing?

Yes, we will release V4L2 this week and the next stage is to do libcamera

**Comment by kbingham** [comment]
Does the V4L2 driver expose a CSI2 receiver as a V4L2 device, and V4L2 subdevices for the sensor? And the ISP as a separate node?

**Comment by MichaelZhuxx** [comment]
> Does the V4L2 driver expose a CSI2 receiver as a V4L2 device, and V4L2 subdevices for the sensor? And the ISP as a separate node?

Yes, we expose sensor, CSI, DVP, ISP as v4l2 subdevice respectively.

**Comment by MichaelZhuxx** [comment]
> Does the V4L2 driver expose a CSI2 receiver as a V4L2 device, and V4L2 subdevices for the sensor? And the ISP as a separate node?

Yes, we expose sensor, CSI, DVP, ISP as v4l2 subdevice respectively.

**Comment by kbingham** [comment]
Great, let me know if there's anything I can do to help then!
Sounds like it might be time to dust off the Beagle-V-CE (collectors edition)

**Comment by MichaelZhuxx** [comment]
@kbingham thanks, definitely need your help to review code at least😄

**Comment by kbingham** [comment]
Sure, please ping me, send me the patches or notify me when they're available.

**Comment by kbingham** [comment]
What's the status on the V4L2 drivers for this ?

**Comment by MichaelZhuxx** [comment]
@kbingham 
Here is pre-release multimedia subsystem for JH7100, including ALSA, V4L2, Gstreamer, VPU decoding. please refer to  [freelight-u-sdk](https://github.com/starfive-tech/freelight-u-sdk/tree/JH7100_starlight_multimedia) and [readme](https://github.com/starfive-tech/freelight-u-sdk/blob/JH7100_starlight_multimedia/README.md) to fetch, build and test it.

next step we will merge multimedia function to linux 5.14.y and keep it up to date

**Comment by esmil** [comment]
@kbingham There is some new multimedia code in the beaglev-5.13.y_multimedia branch. I think testing would be greatly appreciated.
